sim-lib/src/cln.rs

```diff
@@ -101,10 +107,47 @@ impl LightningNode for ClnNode {
 
     async fn track_payment(
         &mut self,
-        _hash: PaymentHash,
-        _shutdown: Listener,
+        hash: PaymentHash,
+        shutdown: Listener,
     ) -> Result<PaymentResult, LightningError> {
-        unimplemented!()
+        loop {
+            tokio::select! {
+                biased;
+                _ = shutdown.clone() => {
+                    break Err(LightningError::TrackPaymentError("Shutdown before tracking results".to_string()));
```

Comment:
Same here, replace by `return`

Reply:
I also think it is worth considering that, if a shutdown signal is raised, we may want to wait until the next tick before stopping the simulation.

This would certainly need a followup, but I think it is worth we discuss it (and we open an issue for it if it makes sense)

Reply:
> wait until the next tick before stopping the simulation

would that be a tick in the current track payment task, i.e 1 second later? how would we "harmonize waiting" for all the other timers and timeouts?

Reply:
What I had in mind is, when a shutdown is triggered:

- First, you stop the producers, so no new payments are passed down the pipeline
- Then you wait for whatever we feel is reasonable for inflight payments to complete

Reply:
concept ack. I created a tracking issue here: https://github.com/bitcoin-dev-project/sim-ln/issues/82